enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/RegisteredRewriter.scala

```diff
@@ -39,6 +39,25 @@ class RegisteredRewriter(tokenContext: TokenContext) {
     val newPipelineInfo = mutable.HashMap[LogicalPlan, PipelineInformation]()
     var rewrites = Map[LogicalPlan, LogicalPlan]()
     val rewritePlanWithRegisters = topDown(Rewriter.lift {
+      /*
+      Projection means executing expressions and writing the result to a row. Since any expression of Variable-type
+      would just write to the row the data that is already in it, we can just skip them
+       */
+      case oldPlan@Projection(_, expressions) =>
+        val information = pipelineInformation(oldPlan)
+        val rewriter = rewriteCreator(information, oldPlan)
+
+        val newExpressions = expressions collect {
+          case (column, expression) if !expression.isInstanceOf[Variable] => column -> expression.endoRewrite(rewriter)
+        }
+
+        val newPlan = oldPlan.copy(expressions = newExpressions)(oldPlan.solved)
```

Comment:
Shouldn't the variables be kept, just not re-written? This code looks like it re-writes the variables away entirely.